sympy/polys/polytools.py

```diff
@@ -52,12 +55,42 @@
 
 
 
+def _polifyit(func):
+    @wraps(func)
+    def wrapper(f, g):
+        g = _sympify(g)
+        if isinstance(g, Poly):
+            return func(f, g)
+        elif isinstance(g, Expr):
+            try:
+                g = f.from_expr(g, *f.gens)
+            except PolynomialError:
+                if g.is_Matrix:
+                    return NotImplemented
+                expr_method = getattr(f.as_expr(), func.__name__)
+                result = expr_method(g)
+                if result is not NotImplemented:
+                    SymPyDeprecationWarning(
```

Comment:
Was it intentional that this only deprecates `Poly*expr` in the case when `expr` cannot be converted to a Poly with the same generators? That is, something like `Poly(x)*x` still works. I guess the difference is that in those cases, it returns a Poly, whereas in the deprecated case here, it returns an Expr.

Reply:
The point is that arithmetic with `Poly` should not implicitly degrade to `Expr`.